project-base/storefront/pages/order-payment-confirmation.tsx

```diff
@@ -1,72 +1,70 @@
 import { MetaRobots } from 'components/Basic/Head/MetaRobots';
 import { CommonLayout } from 'components/Layout/CommonLayout';
-import { PaymentConfirmationContent } from 'components/Pages/Order/PaymentConfirmation/PaymentConfirmationContent';
-import { OrderSentPageContentDocumentApi, useCheckPaymentStatusMutationApi } from 'graphql/generated';
-import { onGtmCreateOrderEventHandler, onGtmPaymentFailEventHandler } from 'gtm/helpers/eventHandlers';
-import { getGtmCreateOrderEventFromLocalStorage } from 'gtm/helpers/helpers';
+import { Webline } from 'components/Layout/Webline/Webline';
+import { PaymentFail } from 'components/Pages/Order/PaymentConfirmation/PaymentFail';
+import { PaymentSuccess } from 'components/Pages/Order/PaymentConfirmation/PaymentSuccess';
+import {
+    usePaymentConfirmationContent,
+    useUpdatePaymentStatus,
+} from 'components/Pages/Order/PaymentConfirmation/helpers';
+import { getStringFromUrlQuery } from 'helpers/parsing/urlParsing';
 import { getServerSidePropsWrapper } from 'helpers/serverSide/getServerSidePropsWrapper';
 import { initServerSideProps, ServerSidePropsType } from 'helpers/serverSide/initServerSideProps';
 import useTranslation from 'next-translate/useTranslation';
 import { useRouter } from 'next/router';
-import { useEffect } from 'react';
+import Skeleton from 'react-loading-skeleton';
+
+const MAX_ALLOWED_PAYMENT_TRANSACTIONS = 2;
 
 const OrderPaymentConfirmationPage: FC<ServerSidePropsType> = () => {
     const { t } = useTranslation();
-    const router = useRouter();
-    const [checkPaymentStatusResult, checkPaymentStatus] = useCheckPaymentStatusMutationApi();
-
-    const { orderIdentifier } = router.query;
-
-    const orderUuidParam = getOrderUuid(orderIdentifier);
+    const { orderIdentifier, orderPaymentStatusPageValidityHash } = useRouter().query;
+    const orderUuid = getStringFromUrlQuery(orderIdentifier);
+    const orderPaymentStatusPageValidityHashParam = getStringFromUrlQuery(orderPaymentStatusPageValidityHash);
 
-    useEffect(() => {
-        checkPaymentStatus({ orderUuid: orderUuidParam }).then(({ data: checkPaymentStatusResultData }) => {
-            const { gtmCreateOrderEventOrderPart, gtmCreateOrderEventUserPart } =
-                getGtmCreateOrderEventFromLocalStorage();
-            if (
-                checkPaymentStatusResultData?.CheckPaymentStatus === undefined ||
-                !gtmCreateOrderEventOrderPart ||
-                !gtmCreateOrderEventUserPart
-            ) {
-                return;
-            }
+    const { isPaymentStatusFetching, paymentStatusData } = useUpdatePaymentStatus(
+        orderUuid,
+        orderPaymentStatusPageValidityHashParam,
+    );
+    const { content, isContentFetching } = usePaymentConfirmationContent(
+        orderUuid,
+        paymentStatusData?.UpdatePaymentStatus,
+    );
```

Comment:
I we can't have those two hooks in one hook? Since they are connected together, it seems to me like it is not necessary to have them separated. I guess it would also simplify the logic here.

Reply:
I combined them and removed isPaymentStatusFetching as it is allways shorter-lived than isContentFetching